translate/convert/po2l20n.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2002-2015 Zuza Software Foundation
+#
+# This file is part of translate.
+#
+# translate is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# translate is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+"""Convert Gettext PO localization files to .l20n files.
+"""
+
+import logging
+import re
+import warnings
+
+from translate.convert import accesskey, convert
+from translate.lang import data
+from translate.misc import quote
+from translate.storage import po, l20n
+
+
+logger = logging.getLogger(__name__)
+
+
+class po2l20n:
+
+    def __init__(self, includefuzzy=False):
+        self.includefuzzy = includefuzzy
+
+    def convertunit(self, unit):
+        l20n_unit = l20n.l20nunit(unit.getlocations()[0])
+        l20n_unit.value = unit.target or unit.source
+        #l20n_unit.comment = unit.allcomments
+
+        return l20n_unit
+
+    def mergestore(self, origl20nfile, translatedl20nfile):
+        target_store = l20n.l20nfile()
+        translatedl20nfile.makeindex()
+        for l20nunit in origl20nfile.units:
+            l20nunit_id = l20nunit.getid()
+            if l20nunit_id in translatedl20nfile.id_index:
+                target_store.addunit(translatedl20nfile.id_index[l20nunit_id])
+
+        return target_store
+
+    def convertstore(self, po_store):
+        outputstore = l20n.l20nfile()
+        for unit in po_store.units[1:]:
+            if not self.includefuzzy and unit.isfuzzy():
+                continue
+            newunit = self.convertunit(unit)
+            if newunit is not None:
+                outputstore.addunit(newunit)
+        return outputstore
+
+
+def convertl20n(inputfile, outputfile, templatefile, includefuzzy=False,
+                outputthreshold=None):
```

Comment:
While I think `convertl20n` is needed as an interface between the `ConvertOptionParser` logic and the actual convert logic for the formats at hand, I think we can move most of this specific logic up to `po2l20n` class because it mostly does boilerplate that is shared with other converters. This will allow in the future to abstract all converter classes to a `BaseConverter` and after `ConvertOptionParser` refactoring the `convertl20n` would be able to be dropped in favor of passing a `BaseConverter` instance directly. So:


```python
class po2l20n(object):

    def __init__(self, input_file, output_file, template_file=None,
                 include_fuzzy=False):
        self.output_file = output_file  # The output is written here.
        self.includefuzzy = includefuzzy

        self.input_store = po.pofile(input_file)  # This is PO.
        self.output_store = l20n.l20nfile(output_file)  # This is l20n.

        if template_file is None:
            self.template_store = None
        else:
            self.template_store = l20n.l20nfile(template_file)  # This is l20n.

    def merge_stores(self):
        # Note this must convert the PO store to l20n before aligning it with
        # the template.

    def convert(self):
        if self.template_store is None:
            self.convert_store()  # Changed method name to make it more readable.
        else:
            self.merge_stores()  # Changed method name to make it more readable.

        self.output_file.write(str(self.output_store))  # Since the store keeps
                                                        # a reference to the
                                                        # output file perhaps
                                                        # this and some code on
                                                        # __init__ above can be
                                                        # simplified.

    
def convertl20n(inputfile, outputfile, templatefile, includefuzzy=False,
                outputthreshold=None):
    po2l20n(inputfile, outputfile, templatefile, includefuzzy).convert()
    return 1
```